grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/Utility.java

```diff
@@ -441,6 +441,15 @@ public static RoleType getNonMetaTopRole(RoleType role){
      * @return true if child is subtype of parent
      */
     public static boolean checkTypesCompatible(Type parent, Type child) {
-        return parent.equals(child) || parent.subTypes().contains(child);
+        return parent.subTypes().contains(child);
```

Comment:
Wouldn't it be more efficient to check if the super type's of `child` contains parent? That would involve serialising a lot less concepts.